_posts/2018-08-10-On-Context-Switching.markdown

```diff
@@ -0,0 +1,190 @@
+---
+
+Programming generally requires getting into a good [flow state][flow] and working on a tricky problem for some time.
+In the last few years, most of my work at Artsy (and in the OSS world) has been less about longer-term building of
+hard things, but working on many smaller tasks across a lot of different contexts.
+
+Somehow, I managed to end up in the top of "most active" [GitHub members][active], I feel like a lot of this is due
+to doing [Open Source by Default][ossd] at Artsy and second to being good at context switching. I want to try and
+talk though some of my techniques for handling context switching, as well as a bit of philosophy around adopting and
+owning your tools.
+
+<!-- more -->
+
+## Shallow and Wide Work
+
+I want to encourage as many people as possible to work on fun, deep-flow projects at Artsy. One of the most
+successful ways of achieving this, that I've found, is to spend most of my time working on shallower tasks. An
+example from my last few weeks is hiring. The communication aspects require dozens of emails and internal updates
+that can't be scheduled into safe blocks. _([Totally related BTW, we're hiring][hiring].)_
+
+For programming work I have a few techniques for trying to accomplish a lot of shallow tasks across many repos.
+
+Start off by making yourself accountable to someone. For my OSS, this tends to either be [setting expectations][exp]
+in README or using [a VISION][vision] file. For Artsy work we have product managers and engineers who own the
+projects I'm contributing to. For this blog post, it's my [buddy Chris][chris].
+
+I would then strive to get _anything_ out, this could be a work-in-progress PR or via declaratively via
+[README-driven-development][rdd]. Part of this is because you might end up being dragged off into something else,
+and another is that you're less likely to grok the domain better than your reviewers. Whilst not every change is an
+improvement, every improvement adds up - even in small increments.
+
+One way to instantly get rich domain knowledge is by pairing with someone who is more involved. This is a perfect
+way to understand how decisions were made and provides great insight into how someone works on a project. While
+pairing, you might also find additional ways to improve the daily workflow for someone else too!
+
+Trying to have a shallow and wide understanding of many systems means accepting that you can't know the finer
+details about how everything works. You want to know when really big interesting things are happening, but most work
+should be iterative and less relevant to external folk like you. One technique I use for keeping track of system
+changes is by making my own streams of updates and not being too concerned about reading every single one of them.
+For my GitHub organisations this means making a slack channel that gets updates for PRs and releases. You can see
+many overviews, but not get bogged down in the useful discussion.
```

Comment:
I think this paragraph outlines a key point to the blog post, but it might not be obvious to people reading. You're describing really high-level concepts, so concrete examples, or even diagrams, might really help drive your point home.

Reply:
Good idea, might delay shipping tomorrow to think about this